wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/AuroraHostListProvider.java

```diff
@@ -207,6 +215,7 @@ protected void init() throws SQLException {
     }
 
     this.isInitialized = true;
+    lock.unlock();
```

Comment:
It might make sense to create a utility lock that implements `Closable` and then wrap this in `try` block, either way it should be in a try block with the unlock in a `finally`